pkg/lib/configreader/bool.go

```diff
@@ -30,21 +30,22 @@ type BoolValidation struct {
 	Required         bool
 	Default          bool
 	TreatNullAsFalse bool            // `<field>: ` and `<field>: null` will be read as `<field>: false`
+	CantBeSpecified  string          // if provided, returns an error with the provided message if the field is specified
```

Comment:
This should be a string pointer. If specified but empty string, can have default message like "can't be specified"